### DISCUSSION TITLE
query spatial point in irregular grid

### DISCUSSION THREAD

**Discussion by raybellwaves** [corteva/rioxarray#270] [Q&A]
I would like to query a spatial point in an xarray object which has dimensions x and y which are a function of lon and lat.

An example is the DataArray provided in the xarray docs: http://xarray.pydata.org/en/stable/plotting.html#multidimensional-coordinates

```
import xarray as xr
import numpy as np

lon, lat = np.meshgrid(np.linspace(-20, 20, 5), np.linspace(0, 30, 4))
lon += lat / 10
lat += lon / 10
da = xr.DataArray(
    np.arange(20).reshape(4, 5),
    dims=["y", "x"],
    coords={"lat": (("y", "x"), lat), "lon": (("y", "x"), lon)},
)
```

I would like to query the point lat = 20 and lon = 10.

I originally asked this in the xarray discussion: https://github.com/pydata/xarray/discussions/5009 and a kdtree solution was proposed. I would like to know if rioxarray can help.

**Comment by snowman2** [comment]
See #202
